### PR TITLE
feat: read in dependency charts

### PR DESCRIPTION
**Pull request by qvalentin**
This allows using the following of the dependency charts:

### template files
- you can use hover/go-to-definition on includes of dependencies e.g. `name: {{ include "common.names.name" . }}`

### values files
- you can use hover/completion/go-to-definition with the values of the dependencies e,g, `example: {{ .Values.common.exampleValue }}`

Todos:

- [x] Add note in readme that you should run `helm dependency build`
- [x] Add dependency charts to the chartstore
- [x] make sure values files are synced to disk  
- [x] tests

Feedback would be welcome.
So if you want to test, clone this branch and run `go build`.

### CODE REVIEW

File: internal/language_features/includes.go [hunk 1]
```diff
@@ -115,8 +123,12 @@ func (f *IncludesFeature) getDefinitionsHover(includeName string) protocol.Hover
 			node := doc.Ast.RootNode().NamedDescendantForPointRange(referenceRange.StartPoint, referenceRange.EndPoint)
 			if node != nil {
 				result = append(result, protocol.HoverResultWithFile{
-					Value: node.Content([]byte(doc.Content)),
-					URI:   doc.URI,
+					Value: fmt.Sprintf(
+						`%shelm
+%s
```
Comment (luisdavim) [inline internal/language_features/includes.go:128]: nit: wouldn't it be better to use double quotes and `\n`?

Comment (qvalentin) [reply internal/language_features/includes.go:128]: I'm not sure how `\n` will behave on windows, I guess I will write a test for it.

**Comment by dapc11** [comment]
Does this include support for hovering and goto definition of TPL functions? From what I have tested it is working with values and dependency charts, but unfortunately not tpl files.. Any ideas?

Thanks for your work!

**Comment by qvalentin** [comment]
Hi @dapc11, this should work, but I only tested it with the bitnami-common chart as a dependecy.
Your example does not even use depdencies, is this not working for you? For me it is:

image

Are you sure, you got the latest version of helm-ls?

**Comment by dapc11** [comment]
Thanks for quick reply, yes I'm using the latest binary. However, you're right about the lack of dependency usage. My assumption is that when the tpl is "chart-local", i.e. part of the same templates folder it would snap it up automatically. But if you got it working it might be on my end that something is wrong. Will spend some additional time and get back to you.

Cheers!

**Comment by qvalentin** [comment]
Do you have a normal helm chart project structure, or are you using something like helmfile? I just deleted the Chart.yaml file in my example and this causes the same issue as you have stated.

**Comment by dapc11** [comment]
It is a complex setup for sure. Could not really disclose it, unfortunately. But, I was able to reproduce with https://github.com/helm/examples, when I first tried to goto definition in _deployment.yaml_:
`   name: {{ include "hello-world.fullname" . }}`
it succeeded as expected. But the second time, now doing it from _serviceaccount.yaml_:
`  name: {{ include "hello-world.serviceAccountName" . }}`, the lsp process died. Can it be related to the template being encapsulated within conditions that the LSP have a hard time dealing with?

Here're the logs from the above test session:

```
[START][2024-09-06 15:20:29] LSP logging initiated
[WARN][2024-09-06 15:20:29] ...lsp/handlers.lua:135	"The language server yamlls triggers a registerCapability handler for workspace/didChangeConfiguration despite dynamicRegistration set to false. Report upstream, this warning is harmless"
[ERROR][2024-09-06 15:20:37] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"helm-lint-langserver: connections opened","time":"2024-09-06T15:20:37+02:00"}\n'
[ERROR][2024-09-06 15:20:37] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"Workspace configuration: {{true *.{yaml,yml} \\u003cbtree:[\\u003csuper\\u003e\\u003c-\\u003ctext:`.`\\u003e-\\u003e\\u003cany_of:[\\u003ctext:`yaml`\\u003e,\\u003ctext:`yml`\\u003e]\\u003e]\\u003e yaml-language-server 50 false {map[kubernetes:templates/**] true true {true https://www.schemastore.org/api/json/catalog.json}}} {values.yaml values.lint.yaml values*.yaml} info}","time":"2024-09-06T15:20:37+02:00"}\n'
[ERROR][2024-09-06 15:20:37] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"Changing workspace config is not implemented","time":"2024-09-06T15:20:37+02:00"}\n'
[ERROR][2024-09-06 15:20:37] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"error","msg":"Error loading values file ~/repos_personal/examples/charts/hello-world/values.lint.yaml open ~/repos_personal/examples/charts/hello-world/values.lint.yaml: no such file or directory","time":"2024-09-06T15:20:37+02:00"}\n'
[ERROR][2024-09-06 15:20:37] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"helm lint: result for chart ~/repos_personal/examples/charts/hello-world : map[~/repos_personal/examples/charts/hello-world/Chart.yaml:[{{{0 0} {0 0}} Information \\u003cnil\\u003e \\u003cnil\\u003e Helm lint icon is recommended [] [] \\u003cnil\\u003e}]]","time":"2024-09-06T15:20:37+02:00"}\n'
[ERROR][2024-09-06 15:20:40] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"helm lint: result for chart ~/repos_personal/examples/charts/hello-world : map[~/repos_personal/examples/charts/hello-world/Chart.yaml:[{{{0 0} {0 0}} Information \\u003cnil\\u003e \\u003cnil\\u003e Helm lint icon is recommended [] [] \\u003cnil\\u003e}]]","time":"2024-09-06T15:20:40+02:00"}\n'
[ERROR][2024-09-06 15:21:06] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"helm-lint-langserver: connections opened","time":"2024-09-06T15:21:06+02:00"}\n'
[ERROR][2024-09-06 15:21:06] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"Workspace configuration: {{true *.{yaml,yml} \\u003cbtree:[\\u003csuper\\u003e\\u003c-\\u003ctext:`.`\\u003e-\\u003e\\u003cany_of:[\\u003ctext:`yaml`\\u003e,\\u003ctext:`yml`\\u003e]\\u003e]\\u003e yaml-language-server 50 false {map[kubernetes:templates/**] true true {true https://www.schemastore.org/api/json/catalog.json}}} {values.yaml values.lint.yaml values*.yaml} info}","time":"2024-09-06T15:21:06+02:00"}\n'
[ERROR][2024-09-06 15:21:07] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"Changing workspace config is not implemented","time":"2024-09-06T15:21:07+02:00"}\n'
[ERROR][2024-09-06 15:21:07] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"error","msg":"Error loading values file ~/repos_personal/examples/charts/hello-world/values.lint.yaml open ~/repos_personal/examples/charts/hello-world/values.lint.yaml: no such file or directory","time":"2024-09-06T15:21:07+02:00"}\n'
[ERROR][2024-09-06 15:21:07] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"helm lint: result for chart ~/repos_personal/examples/charts/hello-world : map[~/repos_personal/examples/charts/hello-world/Chart.yaml:[{{{0 0} {0 0}} Information \\u003cnil\\u003e \\u003cnil\\u003e Helm lint icon is recommended [] [] \\u003cnil\\u003e}]]","time":"2024-09-06T15:21:07+02:00"}\n'
[ERROR][2024-09-06 15:21:07] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"Error calling yamlls for diagnostics write to stream: write data to conn: write /dev/stdout: file already closed","time":"2024-09-06T15:21:07+02:00"}\n'
[ERROR][2024-09-06 15:21:10] .../vim/lsp/rpc.lua:770	"rpc"	"~/.local/share/nvim/mason/bin/helm_ls"	"stderr"	'{"level":"info","msg":"Error calling yamlls for diagnostics write to stream: write data to conn: write /dev/stdout: file already closed","time":"2024-09-06T15:21:10+02:00"}\n'

```

**Comment by qvalentin** [comment]
Really weird, for me it is working without problems with https://github.com/helm/examples. Your logs also don't indicate a crash of helm-ls (there should be a stack trace).

**Comment by dapc11** [comment]
Hmm, something is fishy here. What editor are you using? Mind sharing your setup?
Cheers

**Comment by qvalentin** [comment]
I'm using neovim (v0.10.1), maybe you can test with the minimal config
(https://github.com/mrjosh/helm-ls/blob/master/examples/nvim/init.lua)

```
nvim --clean -u helm-ls/examples/nvim/init.lua
```

I'm also pretty sure that this should work on macOS and linux

**Comment by dapc11** [comment]
Will try and get back. Thanks for pointers!

**Comment by dapc11** [comment]
It is definitely something on my end, both helm example and the more complex project is working as a charm with the example config. Thanks for your help and great contribution @qvalentin!

**Comment by qvalentin** [comment]
Good to know, feel free to ask, if you have any questions about the neovim config.

**Comment by dapc11** [comment]
Funny story actually, have a quite sophisticated neovim setup, but have been trying out emacs lately, so now I'm stuck somewhere in between. Could be a reason for the issues I've had. Anyway great to have verified that it is on my end. Again great work.

**Pull request merged**